### PR TITLE
Search users

### PR DESCRIPTION
**Pull request by skateray17**

### CODE REVIEW

File: src/routes/balance-router.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-
+import authorization from '../controllers/authorization';
 
 const router = express.Router();
 
 router.route('/')
+  .get(authorization.authorizeAdmin)
   .get((req, res) => {
-    balanceController.getBalanceList()
+    balanceController.findUsers(req.query.name, Number.parseInt(req.query.page, 10))
```
Comment (armenbadalyan) [inline src/routes/balance-router.js:10]: parseInt is base 10 by default

File: src/controllers/balance-controller.js [hunk 2]
```diff
@@ -41,10 +39,47 @@ function updateUserBalance(username, balance) {
...
 }
 
-///////////////////////////////////////////////////
-checkUserBalance('Eee').then((res) => {
-  console.log(res);
-});
-checkUserBalance('Aaaaa').then((res) => {
-  console.log(res);
-});
+function seachByNameOrNameAndSurname(arr) {
```
Comment (armenbadalyan) [inline src/controllers/balance-controller.js]: maybe just seachByNameAndSurname ?

File: src/controllers/balance-controller.js [hunk 3]
```diff
@@ -41,10 +39,47 @@ function updateUserBalance(username, balance) {
...
+  if (arr.length >= 3) {
```
Comment (armenbadalyan) [inline src/controllers/balance-controller.js]: What if the name is "Edwin van der Sar" ?

**Review by armenbadalyan — CHANGES_REQUESTED** [review]

File: src/routes/balance-router.js (hunk 1 above)
Comment (skateray17) [reply src/routes/balance-router.js:10]: In some browsers, when string starts with '0', parseInt base is 8, so I prefer always write base 10

Comment (armenbadalyan) [reply src/routes/balance-router.js:10]: True, but in this case we work in a controlled environment (NodeJS)

**Review by armenbadalyan — APPROVED** [review]

**Pull request merged**